pybop/_problem.py

```diff
@@ -1,30 +1,82 @@
 import numpy as np
 
 
-class Problem:
+class BaseProblem:
     """
     Defines a PyBOP single output problem, follows the PINTS interface.
     """
 
     def __init__(
         self,
-        model,
         parameters,
-        dataset,
-        signal="Terminal voltage [V]",
+        model=None,
         check_model=True,
         init_soc=None,
         x0=None,
     ):
         self._model = model
+        if model is not None:
+            self.check_model = check_model
         self.parameters = parameters
-        self.signal = signal
-        self._model.signal = self.signal
-        self._dataset = {o.name: o for o in dataset}
-        self.check_model = check_model
         self.init_soc = init_soc
         self.x0 = x0
         self.n_parameters = len(self.parameters)
+
+        # Set bounds
+        self.bounds = dict(
+            lower=[param.bounds[0] for param in self.parameters],
+            upper=[param.bounds[1] for param in self.parameters],
+        )
+
+        # Sample from prior for x0
+        if x0 is None:
+            self.x0 = np.zeros(self.n_parameters)
+            for i, param in enumerate(self.parameters):
+                self.x0[i] = param.rvs(1)
+        elif len(x0) != self.n_parameters:
+            raise ValueError("x0 dimensions do not match number of parameters")
+
+        # Add the initial values to the parameter definitions
+        for i, param in enumerate(self.parameters):
+            param.update(value=self.x0[i])
+        self.fit_parameters = {o.name: o.value for o in parameters}
+
+        # Then build the model
+
+    def evaluate(self, parameters):
+        """
+        Evaluate the model with the given parameters and return the signal.
+        """
+        raise NotImplementedError
+
+    def evaluateS1(self, parameters):
+        """
+        Evaluate the model with the given parameters and return the signal and
+        its derivatives.
+        """
+        raise NotImplementedError
+
+
+class FittingProblem(BaseProblem):
+    """
+    Defines a PyBOP single output problem, follows the PINTS interface.
+    """
+
+    def __init__(
+        self,
+        parameters,
+        dataset,
+        model=None,
```

Comment:
What happens if we try to use the FittingProblem without a model?

Reply:
Good point. I think that it should be possible to use BaseProblem without a model (i.e. to allow for a StandaloneProblem), but FittingProblem and DesignProblem should probably require a model.